packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts

```diff
@@ -97,20 +97,23 @@ class CfnApiArtifactHandler implements ApiArtifactHandler {
       path.join(resourceDir, stacksDirName, defaultStackName),
     );
 
-    // write the template buffer to the project folder
-    this.writeSchema(path.join(resourceDir, gqlSchemaFilename), serviceConfig.transformSchema);
-
     const authConfig = this.extractAuthConfig(appsyncCLIInputs.serviceConfiguration);
+    const dependsOn = amendDependsOnForAuthConfig([], authConfig);
+    const apiParameters = this.getCfnParameters(serviceConfig.apiName, authConfig, resourceDir);
+    this.ensureCfnParametersExist(resourceDir, apiParameters);
+    this.context.amplify.updateamplifyMetaAfterResourceAdd(category, serviceConfig.apiName, this.createAmplifyMeta(authConfig, dependsOn));
 
-    await this.context.amplify.executeProviderUtils(this.context, 'awscloudformation', 'compileSchema', {
-      resourceDir,
-      parameters: this.getCfnParameters(serviceConfig.apiName, authConfig, resourceDir),
-      authConfig,
-    });
+    if(serviceConfig?.transformSchema) {
+      // write the template buffer to the project folder
```

Comment:
I've squinted at this line for a while, and I can't convince myself that I fully understand the impact (because this is behind the whole service/walkthrough/request generator rigamarole.

If for some reason the schema is not always filled in the same way could this result in us failing to write and compile?

Reply:
When we add/import RDS API, this value in the `serviceConfig` which primarily is associated with `schema.graphql` contents is empty/non-existent and without this additional check that flow fails.
For non RDS flows, we need to assess the impact/DX change.

Reply:
This doesn't have any impact on the DDB scenarios. The existing "Blank Schema" option in add api steps adds the below schema. We are checking whether the schema has a value before compiling it.

```
# This "input" configures a global authorization rule to enable public access to
# all models in this schema. Learn more about authorization rules here: https://docs.amplify.aws/cli/graphql/authorization-rules
input AMPLIFY { globalAuthRule: AuthRule = { allow: public } } # FOR TESTING ONLY!
```